backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala

```diff
@@ -139,6 +156,11 @@ class UserService(
       Fail.Unauthorized(validationErrorMsg).raiseError[ConnectionIO, Unit]
     }
   }
+
+  private def validatePassword(password: String) =
+    UserRegisterValidator
+      .validatePassword(password)
+      .fold(msg => Fail.IncorrectInput(msg).raiseError[ConnectionIO, Unit], _ => ().pure[ConnectionIO])
```

Comment:
We've got the `.fold(...)` in a couple of places now, I'm wondering if there's a way to simplify that. Maybe have a `UserValidator` instead of `UserRegisterValidator` which would accept optional login/email/password and validate the ones that are present? Then we could also have a single method which converts the `Either` into a `Task`.